std/assembly/array.ts

```diff
@@ -462,12 +462,12 @@ export class Array<T> extends ArrayBufferView {
   join(separator: string = ","): string {
     var dataStart = this.dataStart;
     var length = this.length_;
-    if (isString<T>())    return joinStringArray(dataStart, length, separator);
+    if (isString<T>())    return joinStringArray<T>(dataStart, length, separator);
     if (isBoolean<T>())   return joinBooleanArray(dataStart, length, separator);
     if (isInteger<T>())   return joinIntegerArray<T>(dataStart, length, separator);
     if (isFloat<T>())     return joinFloatArray<T>(dataStart, length, separator);
-    if (isArray<T>())     return joinArrays<T>(dataStart, length, separator);
-    if (isReference<T>()) return joinObjectArray<T>(dataStart, length, separator);
+    if (isArray<T>())     return joinReferenceArray<T>(dataStart, length, separator);
+    if (isReference<T>()) return joinReferenceArray<T>(dataStart, length, separator);
```

Comment:
With the change, this looks like it also handles `isArray<T>` above

Reply:
Yes. Previous separate routine was slightly incorrect. For example: [[1,2], [3,4]].join("_") was printed as `1_2_3_4` but should `1,2_3,4`. With current implementation this fixed. And print object is not different to print array.

Probably I could remove `if (isArray<T>) ...` line. Will try